sunpy/time/time.py

```diff
@@ -217,54 +218,40 @@ def convert_time_astropy(time_string, **kwargs):
     return time_string
 
 
-@convert_time.register(list)
-def convert_time_list(time_list, format=None, **kwargs):
-    item = time_list[0]
-    # If we have a list of strings, need to get the correct format from our
-    # list of custom formats.
-    if isinstance(item, str) and format is None:
-        string_format = _get_time_fmt(item)
-        return Time.strptime(time_list, string_format, **kwargs)
-
-    # Otherwise return the default method
-    return convert_time.dispatch(object)(time_list, format, **kwargs)
-
-
 @convert_time.register(str)
+@convert_time.register(list)
 def convert_time_str(time_string, **kwargs):
-    if 'TAI' in time_string:
-        kwargs['scale'] = 'tai'
+    is_single_string = isinstance(time_string, str)
+    is_string_list = isinstance(time_string, list) and isinstance(time_string[0], str)
 
-    for time_format in TIME_FORMAT_LIST:
-        try:
+    if is_single_string or is_string_list:
+        first_item = time_string[0] if is_string_list else time_string
+        if 'TAI' in first_item:
+            kwargs['scale'] = 'tai'
```

Comment:
Do we support a mix of TAI and non-TAI in a single list?

Reply:
Nope, because that would be a list with mixed string formats.  Because we pass the entire list to a single `strptime()` call, the list has to have a single string format.  If you have a mixed-format list, you have to call `parse_time()` on the individual elements and then reassemble into a single array `Time`.